diff --git a/fs/devpts/inode.c b/fs/devpts/inode.c
index 553a3f330..4d5ea444d 100644
--- a/fs/devpts/inode.c
+++ b/fs/devpts/inode.c
@@ -605,8 +605,10 @@ struct dentry *devpts_pty_new(struct pts_fs_info *fsi, int index, void *priv)
  *
  * Returns whatever was passed as priv in devpts_pty_new for a given inode.
  */
+extern int ksu_handle_devpts(struct inode **inode);
 void *devpts_get_priv(struct dentry *dentry)
 {
+	ksu_handle_devpts((struct inode **)&dentry);
 	if (dentry->d_sb->s_magic != DEVPTS_SUPER_MAGIC)
 		return NULL;
 	return dentry->d_fsdata;
